src/main/kotlin/no/fdk/dataservicecatalog/controller/DataServiceController.kt

```diff
@@ -0,0 +1,72 @@
+package no.fdk.dataservicecatalog.controller
+
+import jakarta.validation.Valid
+import no.fdk.dataservicecatalog.domain.FindDataServiceQuery
+import no.fdk.dataservicecatalog.domain.JsonPatchOperation
+import no.fdk.dataservicecatalog.domain.RegisterDataServiceCommand
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.http.ProblemDetail
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.MethodArgumentNotValidException
+import org.springframework.web.bind.annotation.*
+
+@RestController
+@RequestMapping("/catalogs")
+class DataServiceController {
+
+    @GetMapping("{catalogId}/data-services", produces = [MediaType.APPLICATION_JSON_VALUE])
+    fun findDataServicesByCatalogId(@PathVariable catalogId: String): ResponseEntity<List<FindDataServiceQuery>> {
+        return ResponseEntity.status(HttpStatus.NOT_IMPLEMENTED).build()
+    }
+
+    @GetMapping("{catalogId}/data-services/{dataServiceId}", produces = [MediaType.APPLICATION_JSON_VALUE])
+    fun findDataServiceByCatalogIdAndDataServiceId(
+        @PathVariable catalogId: String, @PathVariable dataServiceId: String
+    ): ResponseEntity<FindDataServiceQuery> {
+        return ResponseEntity.status(HttpStatus.NOT_IMPLEMENTED).build()
+    }
+
+    @PostMapping("{catalogId}/data-services", consumes = [MediaType.APPLICATION_JSON_VALUE])
+    fun registerDataServiceByCatalogId(
+        @PathVariable catalogId: String, @Valid @RequestBody registerDataServiceCommand: RegisterDataServiceCommand
```

Comment:
Tviler på at du trenger seperate DTO for create og find
```suggestion
    fun findDataServicesByCatalogId(@PathVariable catalogId: String): ResponseEntity<List<DataService>> {
        return ResponseEntity.status(HttpStatus.NOT_IMPLEMENTED).build()
    }

    @GetMapping("{catalogId}/data-services/{dataServiceId}", produces = [MediaType.APPLICATION_JSON_VALUE])
    fun findDataServiceByCatalogIdAndDataServiceId(
        @PathVariable catalogId: String, @PathVariable dataServiceId: String
    ): ResponseEntity<DataService> {
        return ResponseEntity.status(HttpStatus.NOT_IMPLEMENTED).build()
    }

    @PostMapping("{catalogId}/data-services", consumes = [MediaType.APPLICATION_JSON_VALUE])
    fun registerDataServiceByCatalogId(
        @PathVariable catalogId: String, @Valid @RequestBody dataService: DataService
```